frontend/src/components/CompareAssignmentsTab/CompareAssignmentsTab.js

```diff
@@ -1,46 +1,91 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import { Flex } from '@instructure/ui-flex/lib/Flex';
 
 import Selector from '../Selector/Selector';
 import AssignmentCard from '../VisCards/AssignmentCard';
 
-const CompareAssignmentsTab = () => (
-  <div>
-    <Flex justifyItems="space-between" margin="0 0 medium">
-      <Flex.Item>
-        <Selector options={['Rubic Name', 'that', 'the other']} labelText="Show Rubric:" />
-      </Flex.Item>
-      <Flex.Item>
-        <Selector options={['Due Date', 'another', 'and another']} labelText="Sort:" />
-      </Flex.Item>
-    </Flex>
-    <div className="filter-bar">
-      <Flex>
+import { AppContext } from '../AppState';
+import { pivotHeatMapData } from '../utils';
+
+const CompareAssignmentsTab = () => {
+  const { state, dispatch } = useContext(AppContext);
+
+  // TODO: these transformations and updates to state need to be re-evaluated
```

Comment:
1. We could do the transformation in the reducer when the payload is received, but I was gravitating toward on-demand computation. If the user never visits  the "Compare Assignments" tab, then why should we bother pivoting the data? Having the pivot live in a dispatch in a component make it so it will only happen _if_ this component is ever rendered. But you are right, that data structure may be useful for other visualizations, so perhaps this should be generalized.

2. I suppose it doesn't need to be stored. I was storing it for performance purposes -- I imagine checking the `.length` of a stored array would be quicker than checking the `.length` of a computed array? I like the idea of leveraging `useMemo` to store the computed value based on equality of inputs, but wouldn't that be functionally the same as storing the value in state? For the sake of future a maintainer's cognitive load, I'd lean towards explicitly defining the value placeholder `initialState` and storing the computed value there, but I'm interested in exploring the `useMemo` route.